### PR TITLE
Spreadsheet: Add SplitRange class and CommonRange#filter

### PR DESCRIPTION
**Pull request by u9g**
Syntax: ```R`A0:B`.filter(c => c.value() % 2 === 1);```

### CODE REVIEW

**Comment by u9g** [comment]
I have to rename the commit to CommonRange#filter

File: Base/res/js/Spreadsheet/runtime.js [hunk 1]
```diff
@@ -197,6 +197,18 @@ class CommonRange {
         });
         return new SplitRange(cellNames);
     }
+
+    unique() {
+        const cellNames = [];
+        const values = new Set();
+        this.forEach(cell => {
+            if (!values.has(cell.value())) {
+                values.add(cell.value());
```
Comment (alimpfard) [inline Base/res/js/Spreadsheet/runtime.js:206]: This isn't exactly the cheapest operation around, let's do it once
```suggestion
            const value = call.value();
            if (!values.has(value)) {
                values.add(value);
```

File: Base/res/js/Spreadsheet/runtime.js [hunk 2]
```diff
@@ -197,6 +197,18 @@ class CommonRange {
...
+                cellNames.push(cell.name);
```
Comment (alimpfard) [inline Base/res/js/Spreadsheet/runtime.js:207]: Can we make it so `SplitRange` can take an Array of Positions as well, so we don't have to parse the cell name every time?

File: Base/res/js/Spreadsheet/runtime.js [hunk 3]
```diff
@@ -189,6 +189,40 @@ class CommonRange {
         this.forEach(val => cells.push(val));
         return cells;
     }
+
+    filter(matches) {
+        const cellNames = [];
+        this.forEach(cell => {
+            if (matches(cell)) cellNames.push(cell.name);
```
Comment (alimpfard) [inline Base/res/js/Spreadsheet/runtime.js:196]: same deal with positions here.

File: Base/res/js/Spreadsheet/runtime.js (hunk 1 above)
Comment (u9g) [reply Base/res/js/Spreadsheet/runtime.js]: done :)

File: Base/res/js/Spreadsheet/runtime.js (hunk 3 above)
Comment (u9g) [reply Base/res/js/Spreadsheet/runtime.js]: done :)

File: Base/res/js/Spreadsheet/runtime.js (hunk 2 above)
Comment (u9g) [reply Base/res/js/Spreadsheet/runtime.js]: done :)

**Pull request merged**